packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -1647,6 +1675,11 @@ function getTemplateAttribute(
     );
 
     let attrValue: Literal | Expression;
+
+    // TODO [#3370]: If complex template expressions are adopted, `preparsedJsExpressions`
+    // should be checked. However, to avoid significant complications in the internal types,
+    // arising from supporting both implementations simultaneously, we will re-parse the
+    // expression here when `ctx.config.experimentalComplexExpressions` is true.
     if (isExpression(value) && !escapedExpression) {
         attrValue = parseExpression(ctx, value, location);
```

Comment:
This isn't ideal. The expression at that location has already been parsed (during tokenization) when the flag is enabled. Ideally, we should check the `preparsedJsExpressions` to find the matching one, and use that for the `attrValue`. However, this would require changes to internal types (like that of `attrValue`) that cascade into messy changes in multiple places.

To be clear, these are solely type issues. However, to avoid making messy changes across our internal types, I opted to parse the expression a second time, here. This seems a decent tradeoff for a pilot feature - and can & should be addressed when the flag goes away.

Reply:
I assume this is why `parseExpressionAt` is called twice?

Reply:
Yes.